module/cli_controller.py

```diff
@@ -0,0 +1,136 @@
+        self.target = None
+
+
+    def load_tree(self) -> None:
+        """
+        Loads all files into command structure
+        """
+        for file in self.command_file_list:
+            load_commands(self.root_node, self.global_commands, file)
+
+    @staticmethod
+    def make_path_string(path_list) -> str:
+        """
+        Joins given list and appends a ':'
+        Expects a list
+        """
+        return ' / '.join(path_list) + ":"
+
+    def set_target(self, target) -> None:
+        self.target = target
+
+    def print_help(self, node) -> None:
+        """
+        Prints the info and any children or parameters of the node.
+        """
+        print('( ' + node.name + ' )')
+        print("\tInfo: " + node.command_info)
+        if node.parameter_list:
+            print("\tParameters: (" + (", ".join(node.parameter_list)) + ")")
+        elif len(node) > 0:
+            print("\tPossible commands: {}".format(", ".join(node[n].name.lower() for n in node.keys())))
+        else:
+            print("\tThis function requires no parameters and has no children")
+
+        print("\n\tCurrently selected robot: {}".format(self.target))
+
+    def go_back_in_tree(self) -> bool:
+        """
+        Go back one node in the tree structure. You cant go back when in root
+        """
+        if self.current_node.parent:
+            self.current_node = self.current_node.parent
+            return True
+        return False
+
+
+    def go_to_root(self) -> None:
+        """
+        Go to root in tree structure
+        """
+        self.current_node = self.root_node
+    @staticmethod
+    def ask_input(input_queue: queue.Queue, string="") -> None:
+        """
+        Starts a new thread asking the user for input and writes this input to the given input_queue
+        Optional string for input
+        """
+        input_queue.put(input(string))
+
+    def start_thread(self) -> None:
+        """
+        Starts a new thread to make nonblocking input possible. And get the current location, after restart this is always just root
+        """
+        path_string = self.make_path_string(self.current_node.get_branch_names()) + " "
+        self.input_thread = threading.Thread(target=self.ask_input, args=(self.input_queue, path_string))
+        self.input_thread.daemon = True
+        self.input_thread.start()
+
+    def check_input(self) -> None:
+        """
+        Starts thread asking for input if it is currently not and the input_queue is not filled.
+        Otherwise processes items in the input_queue.
+        """
+        if not self.input_thread.isAlive() and self.input_queue.empty():
+            self.start_thread()
+        elif not self.input_queue.empty():
+            self.input_handler.handle_new_input(self.input_queue.get().split(" "))
+
+
+    def process(self) -> None:
+        """
+        Main loop of the module
+        """
+        while self.comm.has_data():
+            frame = self.comm.get_data()
```

Comment:
If you aren't accepting data, this while loop shouldn't be required.

Reply:
In the future we want to receive frames, but could for now be removed as it is currently not needed.